tests/system/feature_store/test_feature_store.py

```diff
@@ -4813,6 +4813,77 @@ def test_merge_different_number_of_entities(self):
         ).to_dataframe()
         assert_frame_equal(expected_all, df, check_dtype=False)
 
+    @staticmethod
+    def _get_sorted_df(df: pd.DataFrame, sort_column: str):
+        result = (
+            df.reindex(sorted(df.columns), axis=1)
+            .sort_values(by=sort_column)
+            .reset_index(drop=True)
+        )
+        return result
+
+    @pytest.mark.parametrize("engine", ["local", "dask"])
+    def test_parquet_filters(self, engine):
+        parquet_path = os.path.relpath(str(self.assets_path / "testdata.parquet"))
+        df = pd.read_parquet(parquet_path)
+        filtered_df = df.query('department == "01e9fe31-76de-45f0-9aed-0f94cc97bca0"')
+        run_uuid = uuid.uuid4()
+        v3io_parquet_source_path = f"v3io:///projects/{self.project_name}/df_parquet_filtered_source_{run_uuid}.parquet"
+        v3io_parquet_target_path = f"v3io:///projects/{self.project_name}/df_parquet_filtered_target_{run_uuid}"
+        df.to_parquet(v3io_parquet_source_path)
+        parquet_source = ParquetSource(
+            "parquet_source",
+            path=v3io_parquet_source_path,
+            additional_filters=[
+                ("department", "=", "01e9fe31-76de-45f0-9aed-0f94cc97bca0")
+            ],
+        )
+        result = parquet_source.to_dataframe()
+        assert_frame_equal(
+            result.sort_values(by="patient_id").reset_index(drop=True),
+            filtered_df.sort_values(by="patient_id").reset_index(drop=True),
+        )
+        feature_set = fstore.FeatureSet(
+            "parquet-filters-fs", entities=[fstore.Entity("patient_id")]
+        )
+
+        target = ParquetTarget(
+            name="department_based_target",
+            path=v3io_parquet_target_path,
+            partitioned=True,
+            partition_cols=["department"],
+        )
+        feature_set.ingest(source=parquet_source, targets=[target])
+        result = target.as_df(additional_filters=("room", "=", 1)).reset_index(
+            drop=False
+        )
+        # We want to include patient_id in the comparison and to sort both the columns and the values.
```

Comment:
Sort both the columns and the values? That doesn't really make sense.

Reply:
The entity column after reset index go to position number 0. That's why we need to order by column name as well.

Reply:
Do you mean that we sort the columns alphabetically and we sort the rows by the index? If so, that's what it should say.

Reply:
fixed comment:)